cmake/StaticAnalyzers.cmake

```diff
@@ -57,7 +58,8 @@ macro(enable_cpplint)
 endmacro()
 
 macro(enable_clang_format)
-    find_program(OPENASSETIO_CLANGFORMAT_EXE clang-format clang-format-10)
+    find_program(OPENASSETIO_CLANGFORMAT_EXE
+        NAMES clang-format clang-format-10)
```

Comment:
@feltech Did we mean to have different versions for format vs tidy?

Reply:
Capturing the off-line discussions, we decided to simply use un-versioned bin names for both `clang-tidy` and `clang-format` as described in the commit comment.

Reply:
We re-added v12 support as some distros seem to only install with a suffix.............. 🙃